pkg/actors/reminders/reminders_test.go

```diff
@@ -0,0 +1,1747 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reminders
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opencensus.io/stats/view"
+	clocktesting "k8s.io/utils/clock/testing"
+
+	"github.com/dapr/dapr/pkg/actors/internal"
+	"github.com/dapr/dapr/pkg/config"
+	diag "github.com/dapr/dapr/pkg/diagnostics"
+	"github.com/dapr/dapr/pkg/diagnostics/diagtestutils"
+	daprt "github.com/dapr/dapr/pkg/testing"
+)
+
+var startOfTime = time.Date(2022, 1, 1, 12, 0, 0, 0, time.UTC)
+
+const (
+	TestAppID                       = "fakeAppID"
+	TestKeyName                     = "key0"
+	TestPodName                     = "testPodName"
+	TestActorMetadataPartitionCount = 3
+
+	actorTimersLastValueViewName     = "runtime/actor/timers"
+	actorRemindersLastValueViewName  = "runtime/actor/reminders"
+	actorTimersFiredTotalViewName    = "runtime/actor/timers_fired_total"
+	actorRemindersFiredTotalViewName = "runtime/actor/reminders_fired_total"
+	errStateStoreNotConfigured       = `actors: state store does not exist or incorrectly configured. Have you set the property '{"name": "actorStateStore", "value": "true"}' in your state store component file?`
+	localhost                        = "localhost"
+)
+
+func newTestReminders() *reminders {
+	conf := internal.Config{
+		AppID:              TestAppID,
+		PlacementAddresses: []string{"placement:5050"},
+		HostedActorTypes:   []string{"cat"},
+	}
+	opts := internal.RemindersProviderOpts{
+		StoreName: "testStore",
+		Config:    conf,
+	}
+	clock := clocktesting.NewFakeClock(startOfTime)
+	r := NewRemindersProvider(clock, opts)
+	r.SetStateStoreProviderFn(fakeTStore)
+	return r.(*reminders)
+}
+
+func fakeTStore() (internal.TransactionalStateStore, error) {
+	return nil, errors.New(errStateStoreNotConfigured)
+}
+
+var fakeStor internal.TransactionalStateStore
+
+func fakeRealTStore() (internal.TransactionalStateStore, error) {
+	if fakeStor == nil {
+		fakeStor = daprt.NewFakeStateStore()
+	}
+	return fakeStor, nil
+}
+
+func fakeRealTStore2() (internal.TransactionalStateStore, error) {
+	return daprt.NewFakeStateStore(), nil
+}
```

Comment:
Why these two "fake real" (quite the oxymoron!) stores?

Also, why is `fakeStor` initialized lazily? That may cause race conditions

Reply:
Removed one of them - was not required, seems left while testing something.
Removed lazy loading of `fakeStore` too.